core/includes/install.core.inc

```diff
@@ -1071,23 +1071,29 @@ function install_settings_form_submit($form, &$form_state) {
   // Update global settings array and save.
   $db_settings = $form_state['storage']['database'];
   $settings['database'] = array(
-    'value' => $db_settings['driver'] . '://' . rawurlencode($db_settings['username']) . ':' . rawurlencode($db_settings['password']) . '@' . $db_settings['host'] . ($db_settings['port'] ? (':' . $db_settings['port']) : '') . '/' . rawurlencode($db_settings['database']),
+    'value' => array(
+      'database' => $db_settings['database'],
+      'username' => $db_settings['username'],
+      'password' => $db_settings['password'],
+      'host' => $db_settings['host'],
+    ),
     'required' => TRUE,
   );
-  $settings['database_prefix'] = array(
-    'value' => $db_settings['prefix'],
-  );
-  backdrop_rewrite_settings($settings);
-
-  // Check if UTF-8 MB4 support exists in the database and add to settings.php.
-  if (install_verify_database_utf8mb4()) {
-    $settings = array();
-    $settings['database_charset'] = array(
-      'value' => 'utf8mb4',
-      'required' => TRUE,
-    );
```

Comment:
We should keep this check but instead set `$db_settings['charset']` to `utf8` if `install_verify_database_utf8mb4()` fails. Something like this:

```
// If UTF-8 MB4 is not supported, use the old UTF-8 charset.
if (install_verify_database_utf8mb4() === FALSE && $db_settings['charset'] === 'utf8mb4') {
  $db_settings['charset'] = 'utf8';
}
```

Reply:
Done.  Thanks.